.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,31 @@
+# docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+
+# This is the top-level workflow to run on all pull-requests, and invokes
+# several other workflow dependencies
+name: Pull Request
+
+on:
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  # Call auxiliary workflows
+  pre_run:
+    uses: ./.github/workflows/pre-run-check.yml
+  verilator:
+    needs: pre_run
+    uses: ./.github/workflows/build-test-verilator.yml
+  interactive_dbg:
+    needs: pre_run
+    uses: ./.github/workflows/interactive-debugging.yml
+  doc_gen:
+    needs: pre_run
```

Comment:
Again, I don't see why the pre-check needs to gate this step. It's useful to know if the RDL doesn't parse sooner than later.

Reply:
Pushed a commit to apply this feedback.